File: 1 - metodosBusca/python/app/core/busca/status.py
from datetime import datetime

class Status:
    def __init__(self):
        self.nro_visitados = 0
        self.profundidade_max = 0
        self.tam_abertos = 0
        self.tempo_inicio = None
        self.ms = None
        self.resolveu = False

    def set_mostra(self, ms):
        self.ms = ms

    def inicia(self):
        self.nro_visitados = 0
        self.profundidade_max = 0
        self.tempo_inicio = datetime.now()

    def termina(self, resolveu):
        self.resolveu = resolveu
        if self.ms is not None:
            self.ms.para()

    def resolveu(self):
        return self.resolveu

    def get_tempo_decorrido(self):
        agora = datetime.now()
        return (agora - self.tempo_inicio).total_seconds()

    def get_visitados(self):
        return self.nro_visitados

    def get_profundidade(self):
        return self.profundidade_max

    def explorando(self, n, s):
        self.tam_abertos = s
        self.nro_visitados += 1

        if n.get_profundidade() > self.profundidade_max:
            self.profundidade_max = n.get_profundidade()

File: 1 - metodosBusca/cSharp/Status.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace cSharp
{
    /**
    * Contem varias informacoes de status sobre a busca
    * 
    */
    public class Status
    {
        int nroVisitados = 0;
        int profundidadeMax = 0; // a max prox. que a busca foi
        int tamAbertos = 0;
        DateTime tempoInicio;
        MostraStatusConsole ms;
        bool resolveu = false;

        void setMostra(MostraStatusConsole ms)
        {
            this.ms = ms;
        }

        void inicia()
        {
            nroVisitados = 0;
            profundidadeMax = 0;
            tempoInicio = new DateTime();
        }

        void termina(bool resolveu)
        {
            this.resolveu = true;
            if (ms != null)
            {
                ms.para();
            }
        }

        public bool resolvido()
        {
            return resolveu;
        }

        public long getTempoDecorrido()
        {
            DateTime agora = DateTime.Now;
            return (agora - tempoInicio).Ticks;
        }

        public int getVisitados()
        {
            return nroVisitados;
        }

        public int getProfundidade()
        {
            return profundidadeMax;
        }

        /** o algoritmo pegou n para explorar de um total de s */
        public void explorando(Nodo n, int s)
        {
            tamAbertos = s;
            nroVisitados++;

            if (n.getProfundidade() > profundidadeMax)
            {
                profundidadeMax = n.getProfundidade();
            }
        }

    }
}
